Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSMergeRuns.py

```diff
@@ -0,0 +1,285 @@
+# pylint: disable=no-init,invalid-name,too-many-locals
+import mantid.simpleapi as api
+from mantid.api import PythonAlgorithm, AlgorithmFactory, MatrixWorkspaceProperty
+from mantid.kernel import Direction, StringArrayProperty, StringListValidator, V3D
+import numpy as np
+
+
+class DNSMergeRuns(PythonAlgorithm):
+    """
+    Merges given runs into one matrix workspace.
+    This algorithm is written for the DNS @ MLZ,
+    but can be adjusted for other instruments if needed.
+    """
+    properties_to_compare = ['omega', 'slit_i_left_blade_position',
+                             'slit_i_right_blade_position', 'slit_i_lower_blade_position',
+                             'slit_i_upper_blade_position', 'polarisation', 'flipper']
+    sample_logs = properties_to_compare + ['wavelength', 'huber', 'T1', 'T2', 'Tsp', 'Ei']
+    workspace_names = []
+    wavelength = 0
+    xaxis = '2theta'
+    outws_name = None
+
```

Comment:
To my opinion, this should not be a problem. I set the values for these attributes in PyExec. Unfortunately, I must declare these attributes here, otherwise I get the pylint warning 'attribute-defined-outside-init'. Is there any better way to avoid this warning?